extensions/json-language-features/client/src/languageParticipants.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DocumentSelector } from 'vscode-languageclient';
+import { Event, EventEmitter, extensions } from 'vscode';
+
+/**
+ * JSON language participant contribution.
+ */
+interface LanguageParticipantContribution {
+	/**
+	 * The id of the language which participates with the JSON language server.
+	 */
+	languageId: string;
+	/**
+	 * true if the language allows comments and false otherwise.
+	 * TODO: implement server side setting
+	 */
+	comments?: boolean;
+}
+
+export interface LanguageParticipants {
+	readonly onDidChange: Event<void>;
+	readonly documentSelector: DocumentSelector;
+	hasLanguage(languageId: string): boolean;
+	useComments(languageId: string): boolean;
+	dispose(): void;
+}
+
+export function getLanguageParticipants(): LanguageParticipants {
+	const onDidChangeEmmiter = new EventEmitter<void>();
+	let languages = new Set<string>();
+	let comments = new Set<string>();
+
+	function update() {
+		const oldLanguages = languages, oldComments = comments;
+
+		languages = new Set();
+		languages.add('json');
+		languages.add('jsonc');
+		comments = new Set();
+		comments.add('jsonc');
+
+		for (const extension of extensions.all) {
```

Comment:
Should be `extensions.allAcrossExtensionHosts`. To get that API, this is needed in `package.json`:
```
  "enabledApiProposals": [
    "extensionsAny"
  ],
```